examples/controls_example/public/react_controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect } from 'react';
+import { BehaviorSubject, combineLatest, debounceTime, filter, skip } from 'rxjs';
+
+import { OptionsListSearchTechnique } from '@kbn/controls-plugin/common/options_list/suggestions_searching';
+import { OptionsListSortingType } from '@kbn/controls-plugin/common/options_list/suggestions_sorting';
+import {
+  OptionsListSuccessResponse,
+  OptionsListSuggestions,
+} from '@kbn/controls-plugin/common/options_list/types';
+import { buildExistsFilter, buildPhraseFilter, buildPhrasesFilter, Filter } from '@kbn/es-query';
+import { useBatchedPublishingSubjects } from '@kbn/presentation-publishing';
+
+import { isValidSearch } from '../../../../common/options_list/is_valid_search';
+import {
+  getSelectionAsFieldType,
+  OptionsListSelection,
+} from '../../../../common/options_list/options_list_selections';
+import { initializeDataControl } from '../initialize_data_control';
+import { DataControlFactory, DataControlServices } from '../types';
+import { OptionsListControl } from './components/options_list_control';
+import { OptionsListEditorOptions } from './components/options_list_editor_options';
+import { OptionsListStrings } from './options_list_strings';
+import {
+  DEFAULT_SEARCH_TECHNIQUE,
+  MIN_OPTIONS_LIST_REQUEST_SIZE,
+  OPTIONS_LIST_CONTROL_TYPE,
+  OPTIONS_LIST_DEFAULT_SORT,
+} from './constants';
+import { fetchAndValidate$ } from './fetch_and_validate';
+import { OptionsListControlContext } from './options_list_context_provider';
+import { OptionsListControlApi, OptionsListControlState } from './types';
+
+export const getOptionsListControlFactory = (
+  services: DataControlServices
+): DataControlFactory<OptionsListControlState, OptionsListControlApi> => {
+  return {
+    type: OPTIONS_LIST_CONTROL_TYPE,
+    getIconType: () => 'editorChecklist',
+    getDisplayName: OptionsListStrings.control.getDisplayName,
+    isFieldCompatible: (field) => {
+      return (
+        !field.spec.scripted &&
+        field.aggregatable &&
+        ['string', 'boolean', 'ip', 'date', 'number'].includes(field.type)
+      );
+    },
+    CustomOptionsComponent: OptionsListEditorOptions,
+    buildControl: async (initialState, buildApi, uuid, controlGroupApi) => {
+      /** Serializable state - i.e. the state that is saved with the control */
+      const searchTechnique$ = new BehaviorSubject<OptionsListSearchTechnique | undefined>(
```

Comment:
Once https://github.com/elastic/kibana/pull/189128 merges, you will not be able to change state on initialize as it will cause unsaved changes. Instead, set default value when reading from BehaviorSubject if the value is undefined.

Reply:
Hmm... Can we not just change the comparator at that point so that `undefined` is equal to whatever the default value is? It feels cleaner to me do it it on initialize

Reply:
I've added the custom comparators in 6c590a7cd543a3106926f71439c8fa235e2e744c (actually realized `selections` should have had a custom comparator anyway because it's an array and so the default `===` won't work). Let me know what you think - will this solve things?